### PR TITLE
Use phive to install tools in CI.

### PR DESCRIPTION
**Pull request by markstory**
Having to provide the gpgkeys is a bit tedious but I've not found a better way to handle this yet. Adding a required parameter also means this is a breaking change for the CI configuration.

### CODE REVIEW

File: .github/workflows/cs-stan.yml [hunk 1]
```diff
@@ -25,24 +29,42 @@ jobs:
...
+
+    - name: Cache composer dependencies
+      uses: actions/cache@v3
+      with:
+        path: ${{ steps.composer-cache.outputs.dir }}
+        key: ${{ runner.os }}-composer-${{ steps.key-date.outputs.date }}-${{ hashFiles('composer.json') }}-${{ matrix.prefer-lowest }}
+
     - name: Composer install
-      uses: ramsey/composer-install@v2
+      run: composer update
```
Comment (othercorey) [inline .github/workflows/cs-stan.yml]: why change the composer install?

Comment (ADmad) [reply .github/workflows/cs-stan.yml]: ramsey/composer-installer already handles caching.

**Comment by dereuromark** [comment]
It seems the fails in https://github.com/cakephp/cakephp/pull/16997 are caused by this cs-stan.yml and not having it the same version being used as defined in the phive xml config.
We need to fix this.

**Comment by dereuromark** [comment]
TBH I am not sure how valuable such a "cross repo" script is if the actual changes needed per repo are kind of simple enough to copy and paste to keep control over the versions and what gets used in CI.
If we can make this repo work out fair enough, but otherwise maybe we want to un-DRY this.

**Review by LordSimal — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**